myapp/app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,57 @@
+class TasksController < ApplicationController
+
+  before_action :set_task, only: [:show, :edit, :update, :destroy]
+
+     def index
+      @tasks = Task.all
+     end
+
+      def show
+          @task = Task.find(params[:id])
```

Comment:
Since you already added the #set_task on`before_action`, I think you can safely remove this line and similar lines inside #edit, #update and #destroy. 👍🏻

Reply:
Done.